#!/usr/bin/env python
PACKAGE = "rosnav"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# grp_planner = gen.add_group("Planner", type="tab")
# grp_planner.add("max_vel_x", double_t, 0, "Maximum linear velocity", 0.5, 0.0, 10.0)
# grp_planner.add("max_vel_theta", double_t, 0, "Maximum angular velocity", 0.5, 0.0, 10.0)
gen.add("active_agent", str_t, 0, "Active agent", "")
gen.add("max_vel_x", double_t, 0, "Maximum linear velocity", 2.0, 0.0, 10.0)
gen.add("max_vel_x_backwards", double_t, 0, "Maximum linear velocity backwards", 2.0, 0.0, 10.0)
gen.add("max_vel_y", double_t, 0, "Maximum lateral velocity", 0.0, 0.0, 10.0)
gen.add("max_vel_trans", double_t, 0, "Maximum translational velocity", 0.0, 0.0, 10.0)
gen.add("max_vel_theta", double_t, 0, "Maximum angular velocity", 0.3, 0.0, 10.0)
gen.add("use_proportional_saturation", bool_t, 0, "Use proportional saturation", False)

exit(gen.generate(PACKAGE, "rosnav", "RosnavLocalPlannerReconfigure"))